asm-lsp: 0.4.2 -> 0.6.0

szaffarano opened the pull request:
## Description of changes

Updates `asm-lsp` to v0.6.0

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

szaffarano committed:
asm-lsp: 0.4.2 -> 0.6.0

szaffarano commented:
cc @NotAShelf

NotAShelf commented:
I'd prefer contacting upstream to release a new tag rather than using an untagged commit.

NotAShelf commented on pkgs/by-name/as/asm-lsp/package.nix:
```diff
@@ -8,14 +8,14 @@ let
 in
-rustPlatform.buildRustPackage {
+rustPlatform.buildRustPackage rec {
   inherit pname version;
```
Version needs to be updated to `asm-lsp-yyyy-mm-dd-unstable` if you insist on building from an untagged commit

NotAShelf commented on pkgs/by-name/as/asm-lsp/package.nix:
```diff
@@ -26,7 +26,15 @@ rustPlatform.buildRustPackage {
   ];
 
-  cargoHash = "sha256-x8Cj39Wki+pdoNIO8QPGK29KFJrHtDMoZJIXFEldno0=";
+  cargoLock = {
+    lockFile = "${src}/Cargo.lock";
+  };
```
Inline this:

```nix
cargoLock.lockFile = ...
```

NotAShelf reviewed (changes requested):
A few changes needed, should be good to go afterwards.

szaffarano commented:
> I'd prefer contacting upstream to release a new tag rather than using an untagged commit.

Yes, I already contacted them. I am waiting for a response to decide whether to use a new tag or a commit hash and adjust the PR based on your comments.

Thanks

szaffarano commented:
@NotAShelf 
Based on your comments, I updated the PR and also fixed the failing test. The upstream developer told me they plan to release a new version soon, so I changed the commit hash to use the current tagged version and eventually will create another PR once the new one is released.

NotAShelf reviewed (approved):
LGTM, cheers

NotAShelf commented:
Ofborg seems to be failing still, could you resolve the eval error?

NotAShelf commented:
you'll also want to squash all commits into one

```
asm-lsp: 0.4.2 -> 0.6.0
```

szaffarano commented:
> you'll also want to squash all commits into one
> 
> ```
> asm-lsp: 0.4.2 -> 0.6.0
> ```

yep, I was waiting to be sure that everything else was fine to just run a single squash.

Thanks
